async_approach.py

```diff
@@ -0,0 +1,43 @@
+import argparse
+import asyncio
+import aiohttp
+import requests
+import time
+import re
+
+URL = 'https://wallpaperscraft.ru/'
+
+
+def get_image_urls(url, image_format='jpg', image_amount=5):
+    return list(re.findall(r'https:[\S]*\.' + f'{image_format}', requests.get(url).text))[:image_amount]
+
+
+async def download(url):
+    start_time_current = time.time()
+    async with aiohttp.ClientSession() as session:
+        async with session.get(url) as response:
+            text = await response.read()
+            name = re.findall(r".*\/([^\/]+)$", str(url))
+            with open(f'{name[0]}', 'wb') as picture:
+                picture.write(text)
```

Comment:
Что-то сложный способ получения имени вы выбрали?
```suggestion
            _, name = url.rsplit("/", 1)
            with open(f'{name}', 'wb') as picture:
                picture.write(text)
```